rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -205,3 +201,225 @@ impl InteropIdRange {
         self.count
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use distributed_id_allocator::compressor::{NormalizationError, SessionTokenError};
+
+    static SESSION_ID_1: &str = "748540ca-b7c5-4c99-83ff-c1b8e02c09d6";
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_fract() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.set_cluster_capacity(5.5 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_negative() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.set_cluster_capacity(-2 as f64);
+    }
+
+    #[test]
+    fn generate_next_id() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        assert_eq!(compressor.generate_next_id(), -1 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn get_token_invalid_uuid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.get_token(String::from("not_a_uuid")); // Errors at SessionId::from_uuid_string()
+    }
+
+    #[test]
+    fn take_next_range() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        let id_1 = compressor.generate_next_id();
+        let count_expected = 5;
+        for _ in 1..count_expected {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
```

Comment:
can we assert that the token in the range is correct?

Reply:
My understanding for the wasm layer unit tests was that we were not looking to verify core compressor method functionality (which are tested in the Core unit tests), but rather to verify that the method calls and return values work as expected. We can change that strategy, but we'll need significantly more unit testing here in the wasm layer.